src/bp-core/classes/class-bp-email-recipient.php

```diff
@@ -90,7 +90,8 @@ public function __construct( $email_or_user, $name = '' ) {
 				$address = current( $email_or_user );
 			} else {
 				$address = key( $email_or_user );
-				$name    = current( $email_or_user );
+				$user_id = get_user_by( 'email', $address )->ID;
```

Comment:
@kent-carlo I think you need to check $user_id is found or not. if not then use email id as name.